pkg/api/v1beta3/types.go

```diff
@@ -866,6 +866,53 @@ type PodTemplateList struct {
 	Items []PodTemplate `json:"items" description:"list of pod templates"`
 }
 
+// JobSpec is the specification of a job.
+type JobSpec struct {
+	// Completions is the number of desired completions.
+	Completions int `json:"completions" description:"number of times the job has completed successfully (desired or observed, as appropriate)"`
+
+	// Selector is a label query over pods that should match the Completions count.
+	// If Selector is empty, it is defaulted to the labels present on the Pod template.
+	Selector map[string]string `json:"selector,omitempty" description:"label keys and values that must match in order to be controlled by this job, if empty defaulted to labels on Pod template"`
+
+	// TemplateRef is a reference to an object that describes the pod that will be created if
+	// insufficient completions are detected.
+	TemplateRef *ObjectReference `json:"templateRef,omitempty" description:"reference to an object that describes the pod that will be created if insufficient completions are detected"`
+
+	// Template is the object that describes the pod that will be created if
+	// insufficient completions are detected. This takes precedence over a
+	// TemplateRef.
+	Template *PodTemplateSpec `json:"template,omitempty" description:"object that describes the pod that will be created if insufficient completions are detected; takes precendence over templateRef"`
+}
+
+// JobStatus represents the current status of a job.
+type JobStatus struct {
+	// Completions is the number of actual completions.
+	Completions int `json:"completions" description:"most recently oberved number of completions"`
```

Comment:
A UI would want to show a user a list of running, successful, and failed pods.  Is this something that can be reconstructed from `get pods` and an appropriate label match, or is it something the JobStatus needs to contain?